app/models/docker_builder_service.rb

```diff
@@ -102,6 +102,11 @@ def run(push: false, tag_as_latest: false)
     JobExecution.start_job(@execution)
   end
 
+  def tag(tag)
```

Comment:
Since this causes an immediate mutation of the remote registry, this might be worth calling `tag!`